Sources/NIO/ChannelPipeline.swift

```diff
@@ -944,7 +957,163 @@ extension ChannelPipeline {
     /// - returns: A future that will be completed when all of the supplied `ChannelHandler`s were added.
     public func addHandlers(_ handlers: ChannelHandler...,
                             position: ChannelPipeline.Position = .last) -> EventLoopFuture<Void> {
-        return addHandlers(handlers, position: position)
+        return self.addHandlers(handlers, position: position)
+    }
+
+    /// Synchronously adds the provided `ChannelHandler`s to the pipeline in the order given, taking
+    /// account of the behaviour of `ChannelHandler.add(first:)`.
+    ///
+    /// - Important: Must be called on the `EventLoop`.
+    /// - Parameters:
+    ///   - handlers: The array of `ChannelHandler`s to add.
+    ///   - position: The position in the `ChannelPipeline` to add the handlers.
+    /// - Returns: A result representing whether the handlers were added or not.
+    internal func addHandlersSync(_ handlers: [ChannelHandler],
+                                  position: ChannelPipeline.Position) -> Result<Void, Error> {
+        switch position {
+        case .first, .after:
+            return self._addHandlersSync(handlers.reversed(), position: position)
+        case .last, .before:
+            return self._addHandlersSync(handlers, position: position)
+        }
+    }
+
+    /// Synchronously adds a sequence of `ChannelHandlers` to the pipeline at the given position.
+    ///
+    /// - Important: Must be called on the `EventLoop`.
+    /// - Parameters:
+    ///   - handlers: A sequence of handlers to add.
+    ///   - position: The position in the `ChannelPipeline` to add the handlers.
+    /// - Returns: A result representing whether the handlers were added or not.
+    @inlinable
+    internal func _addHandlersSync<Handlers: Sequence>(_ handlers: Handlers,
```

Comment:
This `@inlinable` is pointless: the function is never called from a public context. We can strip it and make this function `private` and we'll be getting the same effect.